CIScripts/Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -0,0 +1,90 @@
+. $PSScriptRoot/PesterLogger.ps1
+
+function New-LogSource {
+    Param([Parameter(Mandatory = $true)] [string] $Path,
+          [Parameter(Mandatory = $false)] [PSSessionT[]] $Sessions)
+
+    return $Sessions | ForEach-Object {
+        @{
+            Session = $_
+            Path = $Path
+        }
+    }
+}
+
+function Invoke-CommandRemoteOrLocal {
+    param([ScriptBlock] $Func, [PSSessionT] $Session, [Object[]] $Arguments) 
+    if ($Session) {
+        Invoke-Command -Session $Session $Func -ArgumentList $Arguments
+    } else {
+        Invoke-Command $Func -ArgumentList $Arguments
+    }
+}
+
+function Get-LogContent {
+    param([System.Collections.Hashtable] $LogSource)
+    $ContentGetterBody = {
+        Param([Parameter(Mandatory = $true)] [string] $From)
+        $Files = Get-ChildItem -Path $From -ErrorAction SilentlyContinue
+        $Logs = @{}
```

Comment:
I don't see a reason for a hashmap here. Why not a list (eg. of tuples/hashmaps)?

Reply:
I don't see any benefit of going with list + pairs here either and I don't think it's that important in this case. Hashmap feels more natural to me personally: each file maps to its content (hashmap) vs there are pairs of files and their contents. Also it guarantees uniqueness, although it's not really necessary in this case.